File: src/content/3.10/code/scala/snippet18.scala
trait Procompose[Q[_, _], P[_, _], A, B]

object Procompose{
  def apply[Q[_, _], P[_, _], A, B, C]
      (qac: Q[A, C])(pcb: P[C, B]): Procompose[Q, P, A, B] = ???
}

File: src/content/3.10/code/ocaml/snippet18.ml
module type Procompose = sig
  type ('a, 'b) p
  type ('a, 'b) q

  type (_, _) procompose =
    | Procompose : (('a, 'c) q -> ('c, 'b) p) -> ('a, 'b) procompose
end
